### ISSUE TITLE
Replace stdenv.mkDerivation with mkDerivation from an overlay for qt

### ISSUE THREAD

**Issue by Alkeryn** [NixOS/nixpkgs#67370]
so let's say there is a package that has issues with Qt since the lattest updates (such as ckb-next)
and i wish to replace its `stdenv.mkDerivation` to just `mkDerivation` from a nix overlay without modifying the original default.nix and have the whole thing called by `libsForQt5.callPackage` instead of regular `callPackage`

How am i supposed to do so ? is it even possible with the current functions ? 
Basically i want to change the way it is declared as a derivation but keep its content ^

**Comment by gloaming** [comment]
You can try replacing `stdenv`: `ckb-next.override { stdenv.mkDerivation = ... ; }`

Is there any reason you don't wish to just make the necessary change in your fork? You'll need to to push the changes upstream anyway.

**Comment by Alkeryn** [comment]
@gloaming this was an example, i already pushed the changes but there could be cases in which i wouldn't want to
And it is mostly so i can have a fix in the meanwhile most of the time since i'm using channels and not a nixpkgs clone as my nixpkgs ^

**Comment by gloaming** [comment]
Closing as I believe the question has been answered.
